docs/sql-performance-tuning.md

```diff
@@ -157,6 +157,18 @@ SELECT /*+ REBALANCE(3, c) */ * FROM t;
 
 For more details please refer to the documentation of [Partitioning Hints](sql-ref-syntax-qry-select-hints.html#partitioning-hints).
 
+## Leveraging Statistics
+Apache Spark's ability to choose the best execution plan among many possible options is determined in part by its estimates of how many rows will be output by every node in the execution plan (read, filter, join, etc.). Those estimates in turn are based on statistics that are made available to Spark in one of several ways:
+
+- **Data source**: Statistics that Spark reads directly from the underlying data source, like the counts and min/max values in the metadata of Parquet files. These statistics are maintained by the underlying data source.
+- **Catalog**: Statistics that Spark reads from the catalog, like the Hive Metastore. These statistics are collected or updated whenever you run [`ANALYZE TABLE`](sql-ref-syntax-aux-analyze-table.html).
+- **Runtime**: Statistics that Spark computes itself at runtime as a job is running.
```

Comment:
I'm curious what statistics are considered runtime? Do we want to add examples here?

Reply:
I believe this concept is specific to adaptive query execution. It was [introduced here][1]. I'll clarify this point accordingly.

[1]: https://github.com/apache/spark/pull/32391/files#diff-c42595588570b41aeb8e33e98553522a53cd46751abfc98b1af3495cf4f5bc15R50-R51

Reply:
I've explained this a bit more here and a few lines below. I've also updated the screenshot in the PR description.

Reply:
Thank you @nchammas , that's very helpful!